src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -843,7 +844,8 @@ private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer
                         }
                     };
                 }
-                else if (typeRight.Kind == DKind.OptionSetValue && typeRight.OptionSetInfo.CanCompareNumeric &&
+                else if (typeRight.Kind == DKind.OptionSetValue &&
+                        (!usePowerFxV1CompatibilityRules || typeRight.OptionSetInfo.CanCompareNumeric) &&
                         (!usePowerFxV1CompatibilityRules || typeRight.OptionSetInfo.CanCoerceFromBackingKind || typeRight.OptionSetInfo.CanCoerceToBackingKind) &&
                         (typeLeft.Kind == DKind.Number || typeLeft.Kind == DKind.Decimal))
```

Comment:
This can be a helper as it shares the same logic with above code.

Reply:
Done.